src/org/labkey/test/components/glassLibrary/grids/GridRow.java

```diff
@@ -0,0 +1,203 @@
+package org.labkey.test.components.glassLibrary.grids;
+
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.labkey.test.components.html.Checkbox;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.labkey.test.util.TestLogger.log;
+
+public class GridRow extends WebDriverComponent<GridRow.ElementCache>
+{
+    final WebElement _el;
+    final WebDriver _driver;
+    final ResponsiveGrid _grid;
+    private Map<String, String> _rowMap = null;
+
+    public GridRow(ResponsiveGrid grid, WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+        _grid = grid;
+    }
+
+    public boolean hasSelectColumn()
+    {
+        return elementCache().selectColumn.isPresent();
+    }
+
+    public boolean isSelected()
+    {
+        return hasSelectColumn() && elementCache().selectCheckbox.isSelected();
+    }
+
+    public GridRow select(boolean checked)
+    {
+        elementCache().selectCheckbox.set(checked);
+        return this;
+    }
+
+    public WebElement getCell(int colIndex)
+    {
+        return Locator.tag("td").index(colIndex).findElement(this);
+    }
+
+    public WebElement getCell(String columnHeader)
+    {
+        return getCell(_grid.getColumnIndex(columnHeader));
+    }
+
+    public boolean contains(Locator loc)
+    {
+        return loc.existsIn(this);
+    }
+
+    private WebElement getLink(String text)
+    {
+        log("seeking link with text [" + text + "]");
+        WebElement link = Locator.linkWithText(text).findElement(getComponentElement());
+        getWrapper().scrollIntoView(link);
+        log("found element with text [" + link.getText() + "]");
+        return link;
+    }
+
+    public void clickLink(String text)
+    {
+        WebElement link = getLink(text);
+        String href = link.getAttribute("href");
+        link.click();
+        log("waiting for url to be: [" + href + "]");
+        WebDriverWrapper.waitFor(()-> getWrapper().getURL().toString().endsWith(href), 1000);
+        getWrapper().shortWait().until(ExpectedConditions.stalenessOf(link));
+    }
```

Comment:
I initially wrote the synchronization/wait-for-url-to-update code to work around the fact that clicking a link doesn't get a page refresh in a single-page app. 
Previously the clickLink() function was on the grid, I wanted to push the implementation down into the row as a way to avoid multiple implementations of the same thing.

Speaking of which, there are several use cases that call for more than just text to get an element in a row or cell- I'd like to add better support for these cases like an overload allowing the test code to pass in a locator.  Tests can still get a row or cell and use it as a search scope that way, so win-win?